File: 2023/advent_of_code/test/q12/q12_test.exs
defmodule AdventOfCode.Q12Test do
  use ExUnit.Case
  doctest AdventOfCode.Q12

  test "Q12 Part 1 Example works" do
    file_stream = File.stream!("./test/q12/q12_example.txt")
    result = AdventOfCode.Q12.part1(file_stream)
    assert result == 21
  end

  test "Q12 Part 1 works" do
    file_stream = File.stream!("./test/q12/q12.txt")
    result = AdventOfCode.Q12.part1(file_stream)
    assert result == 8075
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q12/Q12Test.kt
package q12

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q12Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q12.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(504, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q12.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(500, part2(filePath!!))
    }
}